cmd/manager/main.go

```diff
@@ -332,48 +325,46 @@ func execute() {
 		accessReviewer = rbac.NewPermissiveAccessReviewer()
 	}
 
-	if operator.HasRole(operator.NamespaceOperator, roles) {
-		if err = apmserver.Add(mgr, params); err != nil {
-			log.Error(err, "unable to create controller", "controller", "ApmServer")
-			os.Exit(1)
-		}
-		if err = elasticsearch.Add(mgr, params); err != nil {
-			log.Error(err, "unable to create controller", "controller", "Elasticsearch")
-			os.Exit(1)
-		}
-		if err = kibana.Add(mgr, params); err != nil {
-			log.Error(err, "unable to create controller", "controller", "Kibana")
-			os.Exit(1)
-		}
-		if err = asesassn.Add(mgr, accessReviewer, params); err != nil {
-			log.Error(err, "unable to create controller", "controller", "ApmServerElasticsearchAssociation")
-			os.Exit(1)
-		}
-		if err = kbassn.Add(mgr, accessReviewer, params); err != nil {
-			log.Error(err, "unable to create controller", "controller", "KibanaAssociation")
-			os.Exit(1)
-		}
-
-		// Garbage collect any orphaned user Secrets leftover from deleted resources while the operator was not running.
-		garbageCollectUsers(cfg, managedNamespaces)
+	if err = apmserver.Add(mgr, params); err != nil {
+		log.Error(err, "unable to create controller", "controller", "ApmServer")
+		os.Exit(1)
+	}
+	if err = elasticsearch.Add(mgr, params); err != nil {
+		log.Error(err, "unable to create controller", "controller", "Elasticsearch")
+		os.Exit(1)
+	}
+	if err = kibana.Add(mgr, params); err != nil {
+		log.Error(err, "unable to create controller", "controller", "Kibana")
+		os.Exit(1)
+	}
+	if err = asesassn.Add(mgr, accessReviewer, params); err != nil {
+		log.Error(err, "unable to create controller", "controller", "ApmServerElasticsearchAssociation")
+		os.Exit(1)
+	}
+	if err = kbassn.Add(mgr, accessReviewer, params); err != nil {
+		log.Error(err, "unable to create controller", "controller", "KibanaAssociation")
+		os.Exit(1)
 	}
-	if operator.HasRole(operator.GlobalOperator, roles) {
-		if err = license.Add(mgr, params); err != nil {
-			log.Error(err, "unable to create controller", "controller", "License")
-			os.Exit(1)
-		}
-		if err = licensetrial.Add(mgr, params); err != nil {
-			log.Error(err, "unable to create controller", "controller", "LicenseTrial")
-			os.Exit(1)
-		}
 
-		go func() {
-			time.Sleep(10 * time.Second)         // wait some arbitrary time for the manager to start
-			mgr.GetCache().WaitForCacheSync(nil) // wait until k8s client cache is initialized
-			r := licensing.NewResourceReporter(mgr.GetClient())
-			r.Start(operatorNamespace, licensing.ResourceReporterFrequency)
-		}()
+	// Garbage collect any orphaned user Secrets leftover from deleted resources while the operator was not running.
+	garbageCollectUsers(cfg, managedNamespaces)
```

Comment:
Nit: move the call to `garbageCollectUsers` after adding of all the controllers.